File: ObjC_Example/Native/Views/NativeAdView.m
//
//  NativeAdView.m
//  ObjC_Example
//
//  Copyright (c) 2015年 FAN Communications. All rights reserved.
//

#import "NativeAdView.h"

#import <NendAd/NADNativeViewRendering.h>

@interface NativeAdView () <NADNativeViewRendering>

@property (nonatomic, weak) IBOutlet UILabel *prLabel;
@property (nonatomic, weak) IBOutlet UILabel *shortTextLabel;
@property (nonatomic, weak) IBOutlet UILabel *longTextLabel;
@property (nonatomic, weak) IBOutlet UILabel *promotionNameLabel;
@property (nonatomic, weak) IBOutlet UILabel *promotionUrlLabel;
@property (nonatomic, weak) IBOutlet UILabel *actionButtonTextLabel;
@property (nonatomic, weak) IBOutlet UIImageView *adImageView;
@property (nonatomic, weak) IBOutlet UIImageView *logoImageView;

@end

@implementation NativeAdView

- (void)awakeFromNib
{
    [super awakeFromNib];
    
    self.layer.borderColor = [UIColor darkGrayColor].CGColor;
    self.layer.borderWidth = 1.f;
    
    self.actionButtonTextLabel.layer.borderColor = self.actionButtonTextLabel.textColor.CGColor;
    self.actionButtonTextLabel.layer.borderWidth = 1.f;
    self.actionButtonTextLabel.layer.masksToBounds = YES;
    self.actionButtonTextLabel.layer.cornerRadius = 0.f;
    
    if (@available(iOS 13.0, *)) {
        self.backgroundColor = [UIColor colorNamed:@"NativeAdBackgroundColor"];
        self.prLabel.textColor = [UIColor colorNamed:@"TextColor"];
        self.longTextLabel.textColor = [UIColor colorNamed:@"TextColor"];
        self.shortTextLabel.textColor = [UIColor colorNamed:@"TextColor"];
        self.promotionNameLabel.textColor = [UIColor colorNamed:@"TextColor"];
        self.promotionUrlLabel.textColor = [UIColor colorNamed:@"TextColor"];
    }
}

#pragma mark - NADNativeViewRendering.h

- (UILabel *)prTextLabel
{
    return _prLabel;
}

- (UILabel *)shortTextLabel
{
    return _shortTextLabel;
}

- (UILabel *)longTextLabel
{
    return _longTextLabel;
}

- (UILabel *)promotionNameLabel
{
    return _promotionNameLabel;
}

- (UILabel *)promotionUrlLabel
{
    return _promotionUrlLabel;
}

- (UILabel *)actionButtonTextLabel
{
    return _actionButtonTextLabel;
}

- (UIImageView *)adImageView
{
    return _adImageView;
}

- (UIImageView *)nadLogoImageView
{
    return _logoImageView;
}

@end

File: Swift_Example/Native/Views/NativeAdView.swift
//
//  NativeAdView.swift
//  ObjC_Example
//
//  Copyright © 2015年 FAN Communications. All rights reserved.
//

import UIKit
import NendAd

class NativeAdView: UIView, NADNativeViewRendering {

    @IBOutlet fileprivate weak var nativeAdPrTextLabel: UILabel!
    @IBOutlet fileprivate weak var nativeAdShortTextLabel: UILabel!
    @IBOutlet fileprivate weak var nativeAdLongTextLabel: UILabel!
    @IBOutlet fileprivate weak var nativeAdPromotionNameLabel: UILabel!
    @IBOutlet fileprivate weak var nativeAdPromotionUrlLabel: UILabel!
    @IBOutlet fileprivate weak var nativeAdActionButtonTextLabel: UILabel!
    @IBOutlet fileprivate weak var nativeAdImageView: UIImageView!
    @IBOutlet fileprivate weak var nativeAdLogoImageView: UIImageView!
    
    override func awakeFromNib() {
        super.awakeFromNib()
        
        self.layer.borderColor = UIColor.darkGray.cgColor;
        self.layer.borderWidth = 1;
        
        self.nativeAdActionButtonTextLabel.layer.borderColor = self.nativeAdActionButtonTextLabel.textColor.cgColor;
        self.nativeAdActionButtonTextLabel.layer.borderWidth = 1;
        self.nativeAdActionButtonTextLabel.layer.masksToBounds = true;
        self.nativeAdActionButtonTextLabel.layer.cornerRadius = 0;
        
        if #available(iOS 13, *) {
            self.backgroundColor = UIColor(named: "NativeAdBackgroundColor")
            self.nativeAdPrTextLabel.textColor = UIColor(named: "TextColor")
            self.nativeAdLongTextLabel?.textColor = UIColor(named: "TextColor")
            self.nativeAdShortTextLabel?.textColor = UIColor(named: "TextColor")
            self.nativeAdPromotionNameLabel?.textColor = UIColor(named: "TextColor")
            self.nativeAdPromotionUrlLabel?.textColor = UIColor(named: "TextColor")
        }
    }
    
    // MARK: - NADNativeViewRendering

    func prTextLabel() -> UILabel! {
        return self.nativeAdPrTextLabel
    }
    
    func shortTextLabel() -> UILabel! {
        return self.nativeAdShortTextLabel
    }
    
    func longTextLabel() -> UILabel! {
        return self.nativeAdLongTextLabel
    }
    
    func promotionNameLabel() -> UILabel! {
        return self.nativeAdPromotionNameLabel
    }
    
    func promotionUrlLabel() -> UILabel! {
        return self.nativeAdPromotionUrlLabel
    }
    
    func actionButtonTextLabel() -> UILabel! {
        return self.nativeAdActionButtonTextLabel
    }
    
    func adImageView() -> UIImageView! {
        return self.nativeAdImageView
    }
    
    func nadLogoImageView() -> UIImageView! {
        return self.nativeAdLogoImageView
    }
}
